crates/services/txpool_v2/src/collision_manager/basic.rs

```diff
@@ -75,16 +71,23 @@ impl<S: BasicCollisionManagerStorage> Default for BasicCollisionManager<S> {
 }
 
 impl<S: BasicCollisionManagerStorage> BasicCollisionManager<S> {
-    fn gather_colliding_txs(
+    fn gather_colliding_tx(
         &self,
         tx: &PoolTransaction,
-    ) -> Result<Collisions<S::StorageIndex>, Error> {
-        let mut collisions = Collisions::new();
+    ) -> Result<Option<S::StorageIndex>, Error> {
+        let mut collision: Option<S::StorageIndex> = None;
```

Comment:
Ok I will add the documentation, however I don't like to check the number of collisions depending on being executable or not outside of the `CollisionManager` maybe I can add a new separate method to the collision manager. For me it's not only a detector it should be able to resolve them and be sure that the one left are less worth

Reply:
Done I documented in collision manager and check pool full. I have split the detection and resolution of collisions in two different function and reworked the whole logic. Could be a bit optimised on the collisions resolution iteration but I think it will not be our pain point for now